### PR TITLE
Integration with redux-oidc

### PR DESCRIPTION
**Pull request by wpears**
Some updates for integration with the front-end.

Known issue ticketed in #18 

I've also broken oidc-test last I checked without spending much effort to fix it, since it's usefulness is rapidly being eclipsed by the full setup. Once everything is bug free, it could either be removed (or fixed if we still want to use it for various diagnostics)

### CODE REVIEW

File: keycloak/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,6 @@
-FROM jboss/keycloak:2.2.1.Final
+FROM jboss/keycloak:2.3.0.Final
+
+ENV KEYCLOAK_LOGLEVEL TRACE
```
Comment (hkeeler) [inline keycloak/Dockerfile]: Could this be a runtime envvar, set in Compose?  I'm guessing will want to tune the log level up/down without having to rebuild the image.

File: keycloak/standalone.xml [hunk 2]
```diff
@@ -250,6 +250,10 @@
                 <local-cache name="authorization">
                     <eviction max-entries="100" strategy="LRU"/>
                 </local-cache>
+                <local-cache name="keys">
```
Comment (hkeeler) [inline keycloak/standalone.xml]: Is this `keys` section related to the upgrade to Keycloak 2.3.0?

File: keycloak/standalone.xml [hunk 3]
```diff
@@ -484,6 +488,13 @@
                 <default-provider>${keycloak.jta.lookup.provider:jboss}</default-provider>
                 <provider name="jboss" enabled="true"/>
             </spi>
+            <spi name="publicKeyStorage">
```
Comment (hkeeler) [inline keycloak/standalone.xml]: Is this `publicKeyStorage` section related to the upgrade to Keycloak 2.3.0?

File: keycloak/standalone.xml [hunk 4]
```diff
@@ -394,8 +398,8 @@
         <subsystem xmlns="urn:jboss:domain:undertow:3.0">
             <buffer-cache name="default"/>
             <server name="default-server">
-                <http-listener name="default" socket-binding="http" redirect-socket="https"/>
                 <https-listener name="https" socket-binding="https" security-realm="UndertowRealm"/>
+                <http-listener name="default" socket-binding="http" redirect-socket="https"/>
```
Comment (hkeeler) [inline keycloak/standalone.xml]: Did `default` move as part of the upgrade to Keycloak 2.3.0?  If not, can we move it back above `https`?  I think it'll be easier to update this file across upgrades if we keep everything in order.

File: keycloak/standalone.xml (hunk 2 above)
Comment (wpears) [inline keycloak/standalone.xml]: It is, Docker copied from the default container

File: keycloak/standalone.xml (hunk 3 above)
Comment (wpears) [inline keycloak/standalone.xml]: Yep

**Comment by wpears** [comment]
@awolfe76 building the project is done from the docker-compose in this repo. Could be adjusted in the future.

**Comment by hkeeler** [comment]
>Do we need the docker-compose file here or should we update [the one in platform](https://github.com/cfpb/hmda-platform/blob/4a3e51836aea776949d9aac6011d3be4962f4f51/docker-compose.yml) to include this?

Good question.  I've kept a separate Compose setup in this repo so it didn't interrupt the work in the main repo.  However, once https://github.com/cfpb/hmda-platform-ui/pull/257 is merged, the Compose setup in `hmda-platform` is no longer going to work.

I think we _should_ move this setup over to `hmda-platform`, the question is whether to do it now, or as part of a followup PR.  When we do, we'll likely have to do further adjustments to [`000-default.conf`](https://github.com/cfpb/hmda-platform-auth/blob/master/auth-proxy/000-default.conf), removing the reference to the `echo` test API, which we shouldn't have to run under `hmda-platform`.

**Comment by awolfe76** [comment]
> I think we should move this setup over to hmda-platform, the question is whether to do it now, or as part of a followup PR.

I think later is fine.

**Pull request merged**